pkg/docker/config/config.go

```diff
@@ -357,15 +386,15 @@ func RemoveAuthentication(sys *types.SystemContext, registry string) error {
 		// Special-case the built-in helper for auth files.
 		case sysregistriesv2.AuthenticationFileHelper:
 			_, err = modifyJSON(sys, func(auths *dockerConfigFile) (bool, error) {
-				if innerHelper, exists := auths.CredHelpers[registry]; exists {
+				if innerHelper, exists := auths.CredHelpers[key]; exists {
 					removeFromCredHelper(innerHelper)
 				}
-				if _, ok := auths.AuthConfigs[registry]; ok {
+				if _, ok := auths.AuthConfigs[key]; ok {
 					isLoggedIn = true
-					delete(auths.AuthConfigs, registry)
-				} else if _, ok := auths.AuthConfigs[normalizeRegistry(registry)]; ok {
+					delete(auths.AuthConfigs, key)
+				} else if _, ok := auths.AuthConfigs[normalizeRegistry(key)]; ok {
```

Comment:
Doesn’t this turn `logout quay.io/foo` into an unwanted `logout quay.io`?

(Also if we never used `normalizeRegistry` on the write path (or did we? I haven’t checked now), this might be completely unnecessary from the start.)

Reply:
Yeah I think we can drop this path. I also added a unit test to cover the non removal of `quay.io` if `quay.io/some/path` is being tried to remove.